homeassistant/components/sonos/binary_sensor.py

```diff
@@ -30,13 +30,26 @@ async def async_setup_entry(
 ) -> None:
     """Set up Sonos from a config entry."""
 
-    async def _async_create_entity(speaker: SonosSpeaker) -> None:
+    async def _async_create_battery_entity(speaker: SonosSpeaker) -> None:
         _LOGGER.debug("Creating battery binary_sensor on %s", speaker.zone_name)
         entity = SonosPowerEntity(speaker)
         async_add_entities([entity])
 
+    async def _async_create_mic_entity(speaker: SonosSpeaker) -> None:
+        _LOGGER.debug("Creating microphone binary_sensor on %s", speaker.zone_name)
+        entity = SonosMicrophoneSensorEntity(speaker)
+        async_add_entities([entity])
```

Comment:
```suggestion
        async_add_entities([SonosMicrophoneSensorEntity(speaker)])
```

Saves only a line, but you could avoid the assigment